### ISSUE TITLE
[Bug] mTLS NativeConnectionOptions tonic::transport::Error(Transport, NativeCertsNotFound)

### ISSUE THREAD

**Issue by cawalch** [temporalio/sdk-typescript#1584]
### What are you really trying to do?

Use mTLS within `NativeConnectionOptions` to connect to a cluster

### Describe the bug

Moving from 1.11.3 to 1.11.5 broke the `NativeConnectionOptions` / connect call with a TransportError

```
/app/node_modules/@temporalio/worker/lib/connection.js:58
                throw new core_bridge_1.TransportError(err.message);
                      ^
TransportError: tonic::transport::Error(Transport, NativeCertsNotFound)
    at NativeConnection.connect (/app/node_modules/@temporalio/worker/lib/connection.js:58:23)
```

Able to confirm it's an issue with 1.11.5 after reverting back to 1.11.3

### Minimal Reproduction

**tlsConfig option**

```
connectionOptions.tls = {
      serverNameOverride: config.TEMPORAL_TLS_SERVER_NAME,
      serverRootCACertificate: fs.readFileSync(config.TEMPORAL_TLS_CA),
      clientCertPair: {
        crt: fs.readFileSync(config.TEMPORAL_TLS_CERT),
        key: fs.readFileSync(config.TEMPORAL_TLS_KEY),
      },
    }
```

### Environment/Versions

The application is built and deployed as a docker image

- Docker / `node:22.11.0-bookworm-slim`
- Hosting environment / Ubuntu 22

**Comment by mjameswh** [comment]
The docker image you use, `node:*-slim`, doesn't include root TLS CA certificates. You need to include them, as demonstrated [here](https://docs.temporal.io/develop/typescript/core-application#using-nodeslim-images).

The reason that was working until TS SDK 1.11.3 is that we were previously relying on `rustls`'s built-in CAs rather than the system-native ones, but that was incorrect. This was [fixed in v1.11.5](https://github.com/temporalio/sdk-core/pull/824), which means that the OS is now responsible for managing root CAs, as it should be.

**Comment by dafanasiev** [comment]
@mjameswh client must use CA from `serverRootCACertificate` as root-of-trust (when parameter set explicitly), not a system CA list.
I think that the current behavior of the client is a bug.

workaround for this issue:

```sh
# remove system CA list
rm -f /etc/ssl/certs/ca-certificates.crt

# use own CA
cp my-root-ca.pem /etc/ssl/certs/ca-certificates.crt
```
